### ISSUE TITLE
Lost reading progress upon scan

### ISSUE THREAD

**Issue by valankar** [Kareadita/Kavita#2893]
### What happened?

I added one book to my library, started a scan. I then lost all progress in my read/reading books.

### What did you expect?

Progress not change for pre-exiting books.

### Kavita Version Number - If you don not see your version number listed, please update Kavita and see if your issue still persists.

0.8.0 - Stable

### What operating system is Kavita being hosted from?

Docker (LSIO Container)

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

None

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

### Relevant log output

```shell
Here is a book that I've read:

[Kavita] [2024-04-20 12:20:45.079 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Beginning series update on The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves, Forced: false
[Kavita] [2024-04-20 12:20:45.397 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Processing series The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves
[Kavita] [2024-04-20 12:20:45.402 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Updating 1 volumes on The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves
[Kavita] [2024-04-20 12:20:45.405 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Parsing The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves - Volume -100000
[Kavita] [2024-04-20 12:20:45.408 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Adding new chapter, The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves - Vol -100000 Ch -100000
[Kavita] [2024-04-20 12:20:45.441 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed 2 volumes from The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves where parsed infos were not mapping with volume name                                  [Kavita] [2024-04-20 12:20:45.442 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves - Volume -100000:
[Kavita] [2024-04-20 12:20:45.443 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Volume cleanup code was trying to remove a volume with a file still existing on disk (usually volume marker removed) File: /data/books/Unsorted/The (Honest) Truth About Dishon - Dan Ariely.epub
[Kavita] [2024-04-20 12:20:45.443 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves - Volume 100000: /data/books/Unsorted/The (Honest) Truth About Dishon - Dan Ariely.epub
[Kavita] [2024-04-20 12:20:45.464 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves FolderPath to /data/books/Unsorted
[Kavita] [2024-04-20 12:20:45.465 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves LowestFolderPath to /Unsorted
[Kavita] [2024-04-20 12:20:46.061 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Finished series update on The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves in 982 ms
[Kavita] [2024-04-20 12:20:46.071 +00:00  35] [Debug] API.Services.MetadataService [MetadataService] Processing cover image generation for series: The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves
[Kavita] [2024-04-20 12:20:46.074 +00:00  35] [Debug] API.Services.MetadataService [MetadataService] Generating cover image for /data/books/Unsorted/The (Honest) Truth About Dishon - Dan Ariely.epub
[Kavita] [2024-04-20 12:20:46.285 +00:00  35] [Information] API.Services.MetadataService [MetadataService] Updated covers for The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves in 220 milliseconds
[Kavita] [2024-04-20 12:20:46.294 +00:00  35] [Debug] API.Services.MetadataService Dispatching 3 update events
[Kavita] [2024-04-20 12:20:46.526 +00:00  35] [Information] API.Services.Tasks.Metadata.WordCountAnalyzerService [WordCountAnalyzerService] Updated metadata for The (Honest) Truth About Dishonesty: How We Lie to Everyone – Especially Ourselves in 225 milliseconds

And another:

[Kavita] [2024-04-20 12:20:58.854 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Beginning series update on The Four Pillars of Investing, Forced: false
[Kavita] [2024-04-20 12:20:58.869 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Processing series The Four Pillars of Investing
[Kavita] [2024-04-20 12:20:58.870 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Updating 1 volumes on The Four Pillars of Investing
[Kavita] [2024-04-20 12:20:58.870 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Parsing The Four Pillars of Investing - Volume -100000
[Kavita] [2024-04-20 12:20:58.870 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Adding new chapter, The Four Pillars of Investing - Vol -100000 Ch -100000
[Kavita] [2024-04-20 12:20:58.889 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed 2 volumes from The Four Pillars of Investing where parsed infos were not mapping with volume name
[Kavita] [2024-04-20 12:20:58.889 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed The Four Pillars of Investing - Volume -100000:
[Kavita] [2024-04-20 12:20:58.889 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Volume cleanup code was trying to remove a volume with a file still existing on disk (usually volume marker removed) File: /data/books/Unsorted/William Bernstein - The Four Pillars of Investing, Second Edition_ Lessons for Building a Winning Portfolio-McGraw Hill (2023).epub
[Kavita] [2024-04-20 12:20:58.889 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Removed The Four Pillars of Investing - Volume 100000: /data/books/Unsorted/William Bernstein - The Four Pillars of Investing, Second Edition_ Lessons for Building a Winning Portfolio-McGraw Hill (2023).epub
[Kavita] [2024-04-20 12:20:58.890 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating The Four Pillars of Investing FolderPath to /data/books/Unsorted
[Kavita] [2024-04-20 12:20:58.890 +00:00  35] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating The Four Pillars of Investing LowestFolderPath to /Unsorted                                                                                                                                             [Kavita] [2024-04-20 12:20:58.981 +00:00  35] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Finished series update on The Four Pillars of Investing in 127 ms
[Kavita] [2024-04-20 12:20:58.984 +00:00  35] [Debug] API.Services.MetadataService [MetadataService] Processing cover image generation for series: The Four Pillars of Investing
[Kavita] [2024-04-20 12:20:58.984 +00:00  35] [Debug] API.Services.MetadataService [MetadataService] Generating cover image for /data/books/Unsorted/William Bernstein - The Four Pillars of Investing, Second Edition_ Lessons for Building a Winning Portfolio-McGraw Hill (2023).epub                        [Kavita] [2024-04-20 12:20:59.142 +00:00  35] [Information] API.Services.MetadataService [MetadataService] Updated covers for The Four Pillars of Investing in 158 milliseconds                                                                                                                                 [Kavita] [2024-04-20 12:20:59.143 +00:00  35] [Debug] API.Services.MetadataService Dispatching 3 update events
[Kavita] [2024-04-20 12:20:59.449 +00:00  35] [Information] API.Services.Tasks.Metadata.WordCountAnalyzerService [WordCountAnalyzerService] Updated metadata for The Four Pillars of Investing in 305 milliseconds

The only book I added was:

[Kavita] [2024-04-20 12:20:59.935 +00:00  30] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Beginning series update on Bad Therapy, Forced: false
[Kavita] [2024-04-20 12:20:59.987 +00:00  30] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Processing series Bad Therapy
[Kavita] [2024-04-20 12:20:59.987 +00:00  30] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Updating 1 volumes on Bad Therapy
[Kavita] [2024-04-20 12:20:59.987 +00:00  30] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Parsing Bad Therapy - Volume -100000
[Kavita] [2024-04-20 12:20:59.988 +00:00  30] [Debug] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Adding new chapter, Bad Therapy - Vol -100000 Ch -100000
[Kavita] [2024-04-20 12:21:00.174 +00:00  30] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating Bad Therapy FolderPath to /data/books/Unsorted
[Kavita] [2024-04-20 12:21:00.178 +00:00  30] [Debug] API.Services.Tasks.Scanner.ProcessSeries Updating Bad Therapy LowestFolderPath to /Unsorted
[Kavita] [2024-04-20 12:21:00.249 +00:00  30] [Information] API.Services.Tasks.Scanner.ProcessSeries Linking up External Recommendations new series (if applicable)
[Kavita] [2024-04-20 12:21:00.255 +00:00  30] [Information] API.Services.Tasks.Scanner.ProcessSeries [ScannerService] Finished series update on Bad Therapy in 322 ms
```

### Additional Notes

When I restore from a backup and re-initiate the scan, the problem is repeatable. I again lose all my progress.

Also everything On Deck is gone. The sort order for Item Added is all changed as well.

**Comment by scare376** [comment]
This was likely due to the migrations made in 0.8.0 there was a chance for some reading progress loss due to some major changes. There is a csv file in your config folder with all the reading progress that you can use as reference to fix it.

**Comment by valankar** [comment]
I do see the csv file. Is there no way to automatically apply it? I'm unsure how to understand it. For example:

```
1,Books,The Four Pillars of Investing,0-0,0,"William Bernstein - The Four Pillars of Investing, Second Edition_ Lessons for Building a Winning Portfolio-McGraw Hill (2023).epub",,"/data/books/Unsorted/William Bernstein - The Four Pillars of Investing, Second Edition_ Lessons for Building a Winning Portfolio-McGraw Hill (2023).epub",valankar,1,15,//html[1]/BODY/APP-ROOT[1]/DIV[1]/DIV[1]/DIV[1]/APP-BOOK-READER[1]/DIV[1]/DIV[2]/DIV[1]/DIV[1]/SECTION[1]/H3[5],03/23/2024 13:27:43,04/12/2024 16:53:24
```

How do I interpret it?

**Comment by majora2007** [comment]
There should be headers, but essentially it's Library, The Series, the individual file then the user, the user id (1), page 15, and that last part can be coded into the db directly or just ignored because it's where in the page you are.

**Comment by valankar** [comment]
The page number appears to be a large section in the book. It would be good to know how to get to where in the "page" it is read to. It seems that number is "section" and not "page". Anyhow, it would really be helpful if future changes like this either attempted to restore the location/progress, or at least warn the user when a scan is done. That was a complete surprise to me.

**Comment by scare376** [comment]
The release announcement stated that some data loss was expected. Kavita is still very much a work in progress program. Joe did what he could to mitigate the data loss and created the csv to export the progress.

**Comment by majora2007** [comment]
@valankar This is the reality behind how epubs work, hence what that last part is for, to help get you to the "page" you're on. Epubs are generally one really large page, sometimes a whole chapter per page. Kavita and other readers do magic to make it more accessible.

As mentioned on the changelog under important, I warned of the potential data loss. It's frustrating I know, but I order to move forward I had to make a call. Not everyone lost data, for example I only hit 2 series with partial data loss.
